gost94/src/gost94_core.rs

```diff
@@ -196,6 +198,15 @@ impl<P: Gost94Params> Gost94Core<P> {
     }
 }
 
+#[cfg(feature = "oid")]
+#[cfg_attr(docsrs, doc(cfg(feature = "oid")))]
+impl<P> AssociatedOid for Gost94Core<P>
+where
+    P: Gost94Params,
+{
+    const OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.643.2.2.9");
+}
+
```

Comment:
The problem is that the GOST 34.311-95 is referenced using the `1.2.643.2.2.9` OID rather than individual params OIDs. See, for example, the RFC4490 and its examples. Or likewise the examples from MR 26.2.002-2012.

In the worst case, we can use the `1.2.643.2.2.9` OID for `Gost94Core<CryptoProParam>`.

I'm waiting for @IvashchenkoSerhii  to sched the light on `1.2.804.2.1.1.1.1.2.1` usage.

Reply:
Sigh... The situation is quite annoying. We generally use OIDs for a "complete" function. It's not clear how we can support function which has OID in a parametrized form and separate OIDs for its parameters.

>The problem is that the GOST 34.311-95 is referenced using the 1.2.643.2.2.9 OID rather than individual params OIDs. See, for example, the RFC4490 and its examples. Or likewise the examples from MR 26.2.002-2012.

IIUC in the RFC 4490 case the OID is given for the CryptoPro paramset:
>This function is always used with default parameters id-GostR3411-94-CryptoProParamSet (see Section 8.2 of [[CPALGS](https://www.rfc-editor.org/rfc/rfc4490#ref-CPALGS)]).

Maybe MR 26.2.002-2012 is the same?

Reply:
Yes, the mentioned MR document doesn't specify any other params (or a way to specify them), so only CryptoPro params can be used with `1.2.643.2.2.9`.